### PR TITLE
feature : #11603 Add distSQL syntax:'show sharding algorithms'

### LINKED ISSUE #11603: Add distSQL syntax:'show sharding algorithms'.
Hi community,

When I used distSQL, I found that there was no way to query all the sharding algorithms.

Therefore, I consider Add distSQL syntax: "show sharding algorithms "

exmaple: 

```mysql
SHOW SHARDING ALGORITHMS (FROM schemaName)?
```

### PR DESCRIPTION
**Pull request by lanchengx**
Fixes #11603.

Changes proposed in this pull request:
- Add distSQL syntax:'show sharding algorithms'

### CODE REVIEW

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/rql/show.xml [hunk 1]
```diff
@@ -37,7 +37,11 @@
         <table name="t_order" />
         <schema name="schemaName" start-index="38" stop-index="47"/>
     </show-sharding-table-rules>
-    
+
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/rql/show.xml:40]: Please fix the indentation problem

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java [hunk 2]
```diff
@@ -442,7 +443,10 @@
     
     @XmlElement(name = "show-sharding-broadcast-table-rules")
     private final List<ShowShardingBroadcastTableRulesStatementTestCase> showShardingBroadcastTableRulesTestCase = new LinkedList<>();
-    
+
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java:446]: Please fix the indentation problem

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingAlgorithmQueryResultSet.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/**
+ * Result set for show sharding algorithm.
+ */
+public final class ShardingAlgorithmQueryResultSet implements DistSQLResultSet {
+
+    private Iterator<Entry<String, ShardingSphereAlgorithmConfiguration>> data;
+
```
Comment (RaigorJiang) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingAlgorithmQueryResultSet.java:44]: Keep indents (including blank lines) consistent with the previous one.

**Review by RaigorJiang — APPROVED** [review]

File: docs/document/content/features/dist-sql/syntax/rql/rql-sharding-rule.cn.md [hunk 4]
```diff
@@ -48,6 +51,14 @@ SHOW SHARDING BROADCAST TABLE RULES [FROM schemaName]
 | keyGeneratorType               | 分布式主键生成器类型                   |
 | keyGeneratorProps              | 分布式主键生成器参数                   |
 
+### Sharding Algorithms
+
+| 列     | 说明          |
+| ------| --------------|
+| name  | 分片算法名称    |
+| type  | 分片算法类型    |
+| props | 表分片算法参数  |
```
Comment (menghaoranss) [inline docs/document/content/features/dist-sql/syntax/rql/rql-sharding-rule.cn.md:60]: why is `表分片算法参数 `?

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShardingAlgorithmsStatementAssert.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * Show sharding algorithms statement assert.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ShowShardingAlgorithmsStatementAssert {
+
+    /**
+     * Assert show sharding algorithms statement is correct with expected parser result.
+     *
+     * @param assertContext assert context
+     * @param actual        actual show sharding algorithms statement
```
Comment (menghaoranss) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShardingAlgorithmsStatementAssert.java:40]: There should be only one space between the param and the comment

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingAlgorithmQueryResultSet.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    private Collection<Object> buildTableRowData(final Entry<String, ShardingSphereAlgorithmConfiguration> data) {
+        Collection<Object> result = new LinkedList<>();
+        result.add(data.getKey());
+        result.add(data.getValue().getType());
+        result.add(buildProps(data.getValue().getProps()));
+        return result;
+    }
+    
+    private Object buildProps(final Properties props) {
+        return Objects.nonNull(props) ? props.entrySet().stream().map(entry -> entry.getKey() + "=" + entry.getValue()).collect(Collectors.joining(", ")) : "";
```
Comment (menghaoranss) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingAlgorithmQueryResultSet.java:76]: Please search `PropertiesConverter.java` in shardingsphere-infra

**Comment by lanchengx** [comment]
@menghaoranss Hi , Please take the time to review it ， thanks

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingAlgorithmQueryResultSet.java (hunk 6 above)
Comment (menghaoranss) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingAlgorithmQueryResultSet.java:76]: `buildProps ` may be is duplicate with `PropertiesConverter.java` in shardingsphere-infra .

**Review by menghaoranss — CHANGES_REQUESTED** [review]

**Comment by lanchengx** [comment]
Hi, @menghaoranss 
For this discussion https://github.com/apache/shardingsphere/pull/11609#discussion_r682263953 
I modified the comments for other StatementAssert classes in this package.
otherwise, the buildProps() is replaced.

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**